File: FactoryDesignPattern/Shapes_Example_Python/Shapes/Triangle.py
from Shapes.Shape import Shape

class Triangle(Shape):

  def __init__(self, height: float, base: float):
    self.height = height
    self.base = base

  def area(self) -> float:
    return 0.5 * self.base * self.height

  def describe(self) -> None:
    print(f"This is a Triangle with Height: {self.height}, and Base: {self.base} with Area: {self.area()}")

File: FactoryDesignPattern/Shapes_Example/Triangle.java
package FactoryDesignPattern.Shapes_Example;

public class Triangle extends Shape {
  private double height;
  private double base;

  public Triangle(double height, double base) {
    this.height = height;
    this.base = base;
  }

  @Override
  protected double area() {
    return 0.5 * this.base * this.height;
  }

  @Override
  public void describe() {
    System.out.printf("This is a Triangle with area: %.2f%n", area());
  }
}
